### PR TITLE
Some lint cleaning

### PR DESCRIPTION
**Pull request by TacoTheDank**
#### What is it?
- [ ] Bugfix (user facing)
- [ ] Feature (user facing)
- [x] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR
- Fixed some `Not annotated parameter overrides @NonNull parameter` lints
- Deleted a few unused parameters
- Fixed some other miscellaneous lints regarding Kotlin, unresolved documentation, etc.

#### APK testing 
The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by opusforlife2** [comment]
> Please retry analysis of this Pull-Request directly on SonarCloud.

Wot.

**Comment by litetex** [comment]
> Please retry analysis of this Pull-Request directly on [SonarCloud](https://sonarcloud.io/project/issues?id=TeamNewPipe_NewPipe&pullRequest=8068).

grafik

Task failed successfully I guess. Also there is no retry button xD
Rebased the PR, should hopefully work now...

File: app/src/main/java/org/schabi/newpipe/info_list/dialog/InfoItemDialog.java [hunk 1]
```diff
@@ -290,18 +289,15 @@ public Builder addMarkAsWatchedEntryIfNeeded() {
...
-            return this;
         }
 
         /**
-         * Adds the {@link StreamDialogDefaultEntry.PLAY_WITH_KODI} entry if it is needed.
-         * @return the current {@link Builder} instance
+         * Adds the {@link StreamDialogDefaultEntry#PLAY_WITH_KODI} entry if it is needed.
          */
-        public Builder addPlayWithKodiEntryIfNeeded() {
+        public void addPlayWithKodiEntryIfNeeded() {
```
Comment (litetex) [inline app/src/main/java/org/schabi/newpipe/info_list/dialog/InfoItemDialog.java:297]: This is not how the Builder pattern works (also above)

**Review by litetex — COMMENTED** [review]
Thank you for the PR!

Only minor things I found:

File: app/src/main/java/org/schabi/newpipe/info_list/dialog/InfoItemDialog.java (hunk 1 above)
Comment (TacoTheDank) [reply app/src/main/java/org/schabi/newpipe/info_list/dialog/InfoItemDialog.java:297]: As far as I can tell, it doesn't need to be a Builder method though, as the Builder isn't being modified here. Am I just seeing it wrong?

Edit: Nevermind, I am just smooth-brained lol. Will correct.

**Review by litetex — APPROVED** [review]
LGTM

**Pull request merged**